src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleCheckpoint.cpp

```diff
@@ -197,6 +197,57 @@ static bool stack_trace_precondition(const ObjectSample* sample) {
   return sample->has_stack_trace_id() && !sample->is_dead();
 }
 
+class StackTraceChunkWriter {
```

Comment:
Perhaps, a detailed comment explaining how this is working with the regular stacktrace writer would be good to have here.